package com.edudev.gymapp.di

import com.edudev.gymapp.data.repository.AuthRepository
import com.edudev.gymapp.data.repository.AuthRepositoryImpl
import com.edudev.gymapp.data.repository.ProfileRepository
import com.edudev.gymapp.data.repository.ProfileRepositoryImpl
import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
abstract class RepositoryModule {

    @Binds
    @Singleton
    abstract fun bindAuthRepository(impl: AuthRepositoryImpl): AuthRepository

    @Binds
    @Singleton
    abstract fun bindProfileRepository(impl: ProfileRepositoryImpl): ProfileRepository
}